src/application_config/configuring_base.py

```diff
@@ -21,49 +22,45 @@
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase(ABC):
-    """Base class for all ConfigSection classes"""
+class ConfigSectionBase:
+    """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
```

Comment:
Not really sure what the added docstring means? Is it also something for the docstring, or something internal that might be explained in a comment?

Reply:
As you can see, the class is (currently) not defining anything, so why do we add it here? That is because of line 17.
I thought that adding this would motivate why we have added it. Shall I refer explicitly to TConfigSection? Or remove this change?